tools/omero/omero_basic_upload/omero_import.xml

```diff
@@ -0,0 +1,72 @@
+<tool id="omero_import" name="OMERO Image Import" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="20.01" license="MIT">
+    <description> with omero-py </description>
+    <macros>
+        <token name="@TOOL_VERSION@">5.18.0</token>
+        <token name="@VERSION_SUFFIX@">0</token>
+    </macros>
+    <xrefs>
+        <xref type="bio.tools">omero</xref>
+    </xrefs>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">omero-py</requirement>
+        <requirement type="package" version="21.0.2">openjdk</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        mkdir folder &&
+        #for $image in $folder:
+            ln -s '$image' folder/'$image.element_identifier' &&
+        #end for
+
+        omero import folder -T Dataset:name:$dataset_name
+            -s $omero_host
+            -p $omero_port
+            -u $__user__.extra_preferences.get('omero|username', $test_username)
+            -w $__user__.extra_preferences.get('omero|password', $test_password)
+        > $log &&
+
+        omero logout
+    ]]></command>
+    <inputs>
+        <param name="folder" type="data" format="atsf,scn,ali,labels,ffr,jpeg,ndpi,tf8,pty,fff,mnc,vms,mng,xml,pcx,img,pct,ims,bip,ome.xml,bin,flex,his,hdr,mov,psd,spi,ipl,dv,aiix,j2kr,pst,mod,ome.tif,tif,obf,dib,dic,im3,tga,pbm,c01,crw,mrcs,l2d,seq,mdb,cfg,htm,mvd2,arf,vsi,companion.ome,htd,aim,fts,ndpis,r3d.log,nef,res,jpx,ics,rec,nd2,cr2,dcm,mea,ome,lei,lms,j2k,oib,mtb,ima,ets,wlz,pict,sm3,sm2,nrrd,xdce,acff,al3d,zvi,1sc,xys,tiff,pgm,pcoraw,ppm,ipm,set,cxd,ipw,apl,fake,tnb,txt,xv,tf2,ps,log,zip,epsi,j2ki,dicom,top,msr,frm,hed,gif,dm2,dm3,dm4,zpo,wav,wat,2fl,sdt,liff,hx,pic,ome.btf,am,bmp,pnl,r3d_d3d,jpf,png,tfr,dti,nii.gz,cif,fdf,grey,df3,stk,fli,hdf,btf,stp,ch5,v,sld,ids,dv.log,jpk,mrw,r3d,xlog,ano,jpe,sxm,jpg,vws,raw,czi,spl,avi,ome.tf8,inf,spc,ome.tf2,spe,lsm,afm,lif,naf,afi,inr,lim,nd,tim,html,sif,env,tif,csv,map,nii,gel,ome.tiff,oif,par,amiramesh,pr3,fits,lut,jp2,oir,dat,aisf,zfr,zfp,xqd,eps,xqf,st,nhdr,i2i,thm,exp,svs,mrc" multiple="true" optional="false" label="Images to import into OMERO"/>
```

Comment:
Most of the extensions listed here are not valid Galaxy datatypes (all that are included [here](https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/config/sample/datatypes_conf.xml.sample)). In the following list its those that have a 1 in the beginning:


```
0 1sc
0 2fl
0 acff
0 afi
0 afm
0 aiix
0 aim
0 aisf
0 al3d
0 ali
0 am
0 amiramesh
0 ano
0 apl
0 arf
0 atsf
0 bin
0 bip
0 c01
0 cfg
0 ch5
0 companion.ome
0 cr2
0 crw
0 cxd
0 czi
0 dat
0 dcm
0 df3
0 dib
0 dic
0 dicom
0 dm2
0 dm3
0 dm4
0 dti
0 dv
0 dv.log
0 env
0 epsi
0 ets
0 exp
0 fake
0 fdf
0 fff
0 ffr
0 flex
0 frm
0 fts
0 gel
0 grey
0 hdf
0 hed
0 his
0 htd
0 htm
0 hx
0 i2i
0 ics
0 ids
0 im3
0 ima
0 img
0 ims
0 inf
0 inr
0 ipl
0 ipm
0 ipw
0 j2k
0 j2ki
0 j2kr
0 jpe
0 jpeg
0 jpf
0 jpk
0 jpx
0 l2d
0 labels
0 lei
0 lif
0 liff
0 lim
0 lms
0 log
0 lsm
0 lut
0 map
0 mdb
0 mea
0 mnc
0 mng
0 mod
0 mrcs
0 mrw
0 msr
0 mtb
0 mvd2
0 naf
0 nd
0 nd2
0 ndpis
0 nef
0 nii
0 nii.gz
0 obf
0 oib
0 oif
0 oir
0 ome
0 ome.btf
0 ome.tf2
0 ome.tf8
0 ome.tif
0 ome.xml
0 pcoraw
0 pct
0 pic
0 pict
0 pnl
0 pr3
0 ps
0 pst
0 pty
0 r3d
0 r3d_d3d
0 r3d.log
0 raw
0 rec
0 res
0 sdt
0 seq
0 set
0 sld
0 sm2
0 sm3
0 spc
0 spe
0 spi
0 spl
0 st
0 stk
0 stp
0 sxm
0 tfr
0 tga
0 thm
0 tim
0 tnb
0 v
0 vsi
0 vws
0 wat
0 wlz
0 xdce
0 xlog
0 xqd
0 xqf
0 xv
0 xys
0 zfp
0 zfr
0 zpo
0 zvi
1 avi
1 bmp
1 btf
1 cif
1 csv
1 eps
1 fits
1 fli
1 gif
1 hdr
1 html
1 jp2
1 jpg
1 mov
1 mrc
1 ndpi
1 nhdr
1 nrrd
1 ome.tiff
1 par
1 pbm
1 pcx
1 pgm
1 png
1 ppm
1 psd
1 scn
1 sif
1 svs
1 tf2
1 tf8
1 tif
1 tif
1 tiff
1 top
1 txt
1 vms
1 wav
1 xml
1 zip
```

We need to remove those with 0 here (we can try to add the Galaxy datatypes .. let me know).

Reply:
Ah, I took the data types from another Galaxy tools which is already on the tool shed:
https://github.com/BMCV/galaxy-image-analysis/blob/master/tools/bioformats2raw/bf2raw.xml

Reply:
Would be great if you could fix it there as well :)